### PR TITLE
Add unittest to sort

### PR DESCRIPTION
**Pull request by danghai**
PR to add `unittest` to sort. It has some issues in `merge_sort`, `insertion_sort`, and `selection_sort`. I am going to rewrite these sort

### CODE REVIEW

**Comment by goswami-rahul** [comment]
What kind of issues @danghai ?

**Comment by danghai** [comment]
It cannot run under `unittest`. I think It has something wrong in the code

**Comment by goswami-rahul** [comment]
Ok, I will see to it.

File: sort/bubble_sort.py [hunk 1]
```diff
@@ -20,9 +20,14 @@ def swap(i, j):
...
+    return arr
+
+class TestSuite(unittest.TestCase):
+    """
+        test suite for the function (above)
+    """
+    def test_bubble_sort(self):
+        self.assertEqual([1, 5, 23, 57, 65, 1232],
+                         bubble_sort([1, 5, 65, 23, 57, 1232]))
+if __name__ == "__main__":
```
Comment (christianbender) [inline sort/bubble_sort.py]: Please add a new line between ```if __name__ == "__main__":``` and the class ```TestSuite```

Comment (christianbender) [inline sort/bubble_sort.py]: Please add a new line between ```if __name__ == "__main__":``` and the class ```TestSuite```

File: sort/comb_sort.py [hunk 2]
```diff
@@ -32,10 +32,14 @@ def swap(i, j):
...
+    return arr
+
+class TestSuite(unittest.TestCase):
+    """
+        test suite for the function (above)
+    """
+    def test_comb_sort(self):
+        self.assertEqual([1, 5, 23, 57, 65, 1232],
+                         comb_sort([1, 5, 65, 23, 57, 1232]))
+if __name__ == "__main__":
```
Comment (christianbender) [inline sort/comb_sort.py]: The same thing as above.

File: sort/counting_sort.py [hunk 3]
```diff
@@ -22,25 +23,27 @@ def counting_sort(arr):
...
+class TestSuite(unittest.TestCase):
+    """
+        test suite for the function (above)
+    """
+    def test_counting_sort(self):
+        self.assertEqual([1, 5, 23, 57, 65, 1232],
+                         counting_sort([1, 5, 65, 23, 57, 1232]))
+        self.assertEqual([-1232, -65, -57, -23, -5, -1],
+                         counting_sort([-1, -5, -65, -23, -57, -1232]))
 if __name__ == "__main__":
```
Comment (christianbender) [inline sort/counting_sort.py]: Please add a new line , too.

**Comment by christianbender** [comment]
@danghai @goswami-rahul 
I ran ```merge_sort.py``` and get:
> [1, 5, 7, 4, 3, 2, 1, 9, 0, 10, 43, 64]
Traceback (most recent call last):
  File "/home/christian/Programmierung/GitHub/algorithms/sort/merge_sort.py", line 40, in <module>
    print(merge_sort(array))
  File "/home/christian/Programmierung/GitHub/algorithms/sort/merge_sort.py", line 10, in merge_sort
    left, right = merge_sort(arr[mid:]), merge_sort(arr[:mid])
TypeError: slice indices must be integers or None or have an __index__ method

**Comment by christianbender** [comment]
@danghai @goswami-rahul 
The insertion sort don't work because our old friend ```xrange```.

**Comment by goswami-rahul** [comment]
We got some conflicts, which I think are due to new changes to master by @christianbender . These need to be resolved before merge

**Comment by danghai** [comment]
@goswami-rahul  do you fix the new line?

**Comment by christianbender** [comment]
@goswami-rahul sorry. should we merge?

**Comment by goswami-rahul** [comment]
@christianbender no need for sorry :) It happens every time. And yes, I think we can merge now .

**Pull request merged**